aten/src/ATen/native/Unique.cpp

```diff
@@ -154,6 +154,28 @@ std::tuple<Tensor, Tensor, Tensor> _unique_dim_cpu_template(
     const bool consecutive,
     const bool return_inverse,
     const bool return_counts) {
+
+    auto sizes = self.sizes().vec();
+    // check how many zero dimentions exist
+    auto num_zero_dims = std::count(sizes.begin(), sizes.end(),0);
+    
+    // tensor is not well formed as it has 0 sized dimentions
+    if (self.size(dim)==0){
+      AT_CHECK(
+          num_zero_dims == 1,
+          "Number of zero sized dimentions is more than one, so unique cannot be applied ")
+      Tensor output = at::empty({0}, self.options());
+      Tensor inverse_indices =
+          at::empty(self.size(dim), self.options().dtype(kLong));
+      Tensor counts = at::empty({0}, self.options().dtype(kLong));
+
+      return std::make_tuple(output, inverse_indices, counts);
+    }
+    if (self.size(dim) != 0 ){
```

Comment:
Isn't `self.size(dim) != 0` always true here?

Reply:
@codexetreme How about this? You are basically checking a condition that is always true.